backend/src/routes/Passwordless.ts

```diff
@@ -0,0 +1,109 @@
+import express, { Router } from 'express';
+import * as z from 'zod';
+import type { User, Auth0, PasswordlessRequest } from '../services';
+import { normalizePhoneNumber } from '../util/country';
+import { validateRequest } from './helpers/validation';
+
+const PASSWORDLESS_REQUEST_INTERVAL_MILLIS = 1000 * 60 * 2;
+
+const BEGIN_SCHEMA = z.object({
+  body: z.object({
+    phoneNumber: z.string(),
+  }),
+});
+
+const LOGIN_SCHEMA = z.object({
+  body: z.object({
+    phoneNumber: z.string(),
+    code: z.string(),
+  }),
+});
+
+function PasswordlessRoutes(
+  userService: typeof User,
+  auth0Service: typeof Auth0,
+  passwordlessRequestService: typeof PasswordlessRequest
+): Router {
+  const router = express.Router();
+
+  router.post(
+    '/begin',
+    validateRequest(BEGIN_SCHEMA, async (parsedReq, res, req) => {
+      const { id } = req.user;
+      if (!id) {
+        req.log.error('wtf something is going wrong');
+      }
+      const passwordlessRequests = await passwordlessRequestService.getPasswordlessRequestsByUserId(
+        id
+      );
+      const lastRequest = passwordlessRequests[passwordlessRequests.length - 1];
+      if (
+        lastRequest &&
+        Number(new Date()) - lastRequest.requestTime <=
```

Comment:
Nit: Hm, it is a little surprising that this works, but I can verify that it does indeed work. Maybe we should be consistent and cast both to `Number` just for consistency?

Reply:
Sure. requestTime is a Date type in the model so it also returns a javascript date.